com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ColladaUtil.java

```diff
@@ -777,60 +825,81 @@ public boolean equals(Object o) {
                 mesh_index_list.add(index);
             }
         }
-        List<Rig.MeshVertexIndices> mesh_vertex_indices = new ArrayList<Rig.MeshVertexIndices>(mesh.triangles.count*3);
-        for (int i = 0; i < shared_vertex_indices.size() ; ++i) {
-            Rig.MeshVertexIndices.Builder b = Rig.MeshVertexIndices.newBuilder();
-            MeshVertexIndex ci = shared_vertex_indices.get(i);
-            b.setPosition(ci.position);
-            b.setTexcoord0(ci.texcoord0);
-            b.setNormal(ci.normal);
-            mesh_vertex_indices.add(b.build());
+
+        int vertex_count = shared_vertex_indices.size();
+
+        ArrayList<Integer> bone_indices_list = new ArrayList<Integer>(vertex_count*4);
+        ArrayList<Float> bone_weights_list = new ArrayList<Float>(vertex_count*4);
+        int max_bone_count = loadVertexWeights(collada, bone_weights_list, bone_indices_list);
+
+        // Bake the values again into our format
+        Float baked_position_list[] = new Float[vertex_count*3];
+        Float baked_normal_list[] = new Float[0];
+        Float baked_texcoord_list[] = new Float[0];
+        Integer baked_bone_indices_list[] = new Integer[0];
+        Float baked_bone_weights_list[] = new Float[0];
+
+        if (normal_list.size() > 0) {
+            baked_normal_list = new Float[vertex_count*3];
+        }
+        if (texcoord_list.size() > 0) {
+            baked_texcoord_list = new Float[vertex_count*2];
+        }
+        if (bone_indices_list.size() > 0) {
+            baked_bone_indices_list = new Integer[vertex_count*4];
+            baked_bone_weights_list = new Float[vertex_count*4];
         }
 
-        Rig.IndexBufferFormat indices_format;
-        ByteBuffer indices_bytes;
-        if(shared_vertex_indices.size() <= 65536)
-        {
-            // if we only need 16-bit indices, use this primarily. Less data to upload to GPU and ES2.0 core functionality.
-            indices_format = Rig.IndexBufferFormat.INDEXBUFFER_FORMAT_16;
-            indices_bytes = ByteBuffer.allocateDirect(mesh_index_list.size() * 2);
-            indices_bytes.order(ByteOrder.LITTLE_ENDIAN).asShortBuffer();
-            for (int i = 0; i < mesh_index_list.size();) {
-                indices_bytes.putShort(mesh_index_list.get(i++).shortValue());
+        for (int index : mesh_index_list) {
+            MeshVertexIndex ci = shared_vertex_indices.get(index);
```

Comment:
Our previous struct was optimized for reconstructing meshes at runtime.
But we want them explicitly laid out in their lists.